### PR TITLE
refactor: Integrate source link container into measurements container in Examples

### PR DESCRIPTION
**Pull request by andiwand**
The source link container seems not to carry any additional information compared to the measurement container. The only important difference between them is that the source link container is sorted by geometry. I integrated a sorted container into the measurement container, even though this might not be strictly necessary it plays nicer with the existing functionality of accessing such a sorted container.

### CODE REVIEW

**Comment by andiwand** [comment]
cc @stephenswat @paulgessinger @benjaminhuth

File: Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp [hunk 1]
```diff
@@ -698,4 +694,54 @@ ProcessCode TrackFindingAlgorithm::finalize() {
...
+// TODO this is somewhat duplicated in AmbiguityResolutionAlgorithm.cpp
+// TODO we should make a common implementation in the core at some point
+void TrackFindingAlgorithm::computeSharedHits(
+    TrackContainer& tracks, const MeasurementContainer& measurements) const {
+  // Compute shared hits from all the reconstructed tracks
+  // Compute nSharedhits and Update ckf results
+  // hit index -> list of multi traj indexes [traj, meas]
+
+  std::vector<std::size_t> firstTrackOnTheHit(
+      measurements.size(), std::numeric_limits<std::size_t>::max());
```
Comment (paulgessinger) [inline Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp:706]: Does this only use the total number of measurements and nothing else?

**Review by paulgessinger — APPROVED** [review]
Ok. In principle, the idea was to have an example of how to work with source links in the examples, but I concede that what we currently had in there wasn't a particularly good example overall.

**Comment by andiwand** [comment]
> Ok. In principle, the idea was to have an example of how to work with source links in the examples, but I concede that what we currently had in there wasn't a particularly good example overall.

I think the changed version is not much better in that regards but I would have to touch much more code to get there. My original goal was to remove the collections of source links as a whole but we rely a lot on the geometry multimap thingy.

File: Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp (hunk 1 above)
Comment (andiwand) [reply Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp:706]: I believe so. Before that it was the size of the source link container which is the same. The source link index will then just point into this array.

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [dd49200b5a8c704849cd63b8cfc93bd5f0cdb8f2]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719184)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719183)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719182)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719181)||
|:yellow_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719180)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719179)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719178)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719177)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719176)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719175)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719174)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719173)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719172)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719171)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719170)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719169)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719168)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719167)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719166)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719165)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719164)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719163)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719162)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719161)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719160)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719159)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719158)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719157)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719156)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719155)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719154)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719153)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44719152)||

**Pull request merged**